### DISCUSSION TITLE
[CodeMaidIgnore] Attribute

### DISCUSSION THREAD

**Discussion by gkowieski** [codecadwallader/codemaid#950] [Q&A]
Maybe its there already, but I can't find it if it is.  But it would be very useful to have the ability to decorate a member with an attribute that indicates to CodeMaid to ignore the member and its members.

Thanks for a life saving extension!

**Comment by codecadwallader** [comment]
Thanks for the suggestion!  We don't currently have an attribute like that, but it has come up a few times in the past as a possible way to opt out of CodeMaid activities.  One question I haven't been clear on is how does that get treated relative to other code elements.  If for example Function A is below "ignored" Function B, on a reorganize should(tm) Function A move above Function B or is Function B supposed to be completely invisible like commented out code?

**Comment by gkowieski** [comment]
In my scenario, I typically #region those members I want to not be manipulated.  So for me, I would decorate every member that I want to be omitted from the codemaid logic within my #region.  And I would expect the region to stay together but be sorted below any other logic.  Or, as in my example, maybe let the coder tell you what they want.

Ex:
#region Properties
[CodeMaid(exclude:true, move:false)]
[Order(1)
Prop 1

[CodeMaid(exclude:true, move:false)]
[Order(2)
Prop 1
#end region

#region Constructors
Public MyApp()
{}

Public MyApp(int blah)
{}
#end region


Becomes:
#region Constructors
Public MyApp()
{}

Public MyApp(int blah)
{}
#end region

#region Properties
[CodeMaid(exclude:true, move:false)]
[Order(1)
Prop 1

[CodeMaid(exclude:true, move:false)]
[Order(2)
Prop 1
#end region



Vs (assuming props set to sort first):
#region Properties
[CodeMaid(exclude:true, move:true)]
[Order(1)
Prop 1

[CodeMaid(exclude:true, move:true)]
[Order(2)
Prop 1
#end region

#region Constructors
Public MyApp()
{}

Public MyApp(int blah)
{}
#end region



For me, it would be a lot like my logging attribute I use for my own logger.  If the class is decorated, then all members of the decorated member are ignored.  And so the logic goes into the members of members.

Now that’s out of the way, May I commend you on a fantastic job of creating this beast!  I have clients that have had me create similar “fuzzy” logic solutions and I almost pull my hair out in the various spaghetti loops they create.  So my hats off to you and your crew!  Very well done and the GUI isn’t so strung out we can’t figure it out without documentation.

Even in my own example I see so many different ways it could be interpreted.  So… Good luck!!!!   😁

Thanks,


***@***.***

Greg Kowieski
SR Developer/Technical Analyst.
214-296-9572 x103
www.theworldon-line.com

Privileged and Confidential
The information in this transmission is privileged and confidential and intended only for the recipient listed above. If you are not the intended recipient, please advise the sender immediately by reply e-mail and delete this message and any attachments without retaining a copy. If you are not the intended recipient, you are hereby notified that any disclosure, copying or distribution of this message, or the taking of any action based upon it, is strictly prohibited.
